js/os.ts

```diff
@@ -33,14 +33,22 @@ function setEnv(key: string, value: string): void {
  * the process. The environment object will only accept `string`s
  * as values.
  *
+ *       console.log(Deno.env("SHELL"));
  *       const myEnv = Deno.env();
  *       console.log(myEnv.SHELL);
  *       myEnv.TEST_VAR = "HELLO";
  *       const newEnv = Deno.env();
  *       console.log(myEnv.TEST_VAR == newEnv.TEST_VAR);
  */
-export function env(): { [index: string]: string } {
+export function env(): { [index: string]: string };
+export function env(key: string): string | undefined;
+export function env(
+  key?: string
+): { [index: string]: string } | string | undefined {
   const env = sendSync(dispatch.OP_ENV);
+  if (key) {
+    return env[key];
```

Comment:
Something to consider is that on windows environment variables are case-insensitive (or technically, case-preserving). This is not a theoretical concern, e.g on some systems the path environment variable will be "Path", sometimes it is "PATH". Standard library functions like `getenv()` and `GetEnvironmentVariable()` honor this and perform case-insensitive lookups. So I think Deno should too.

Reply:
Should this only happen on Windows or should it be on Mac/Linux too?

Reply:
Only windows is case-insensitive.
Linux and OS X are case-sensitive.

Reply:
Do we need a new op to get a single env var? Is `key.toUpperCase()` sufficient? Please advise @piscisaureus.

Reply:
> Is key.toUpperCase() sufficient?  

It might be, but I don't know what collation and/or what version of unicode windows uses.
A simple test I did revealed 311 characters for which `toUpperCase()` would not do the right thing.
Using `toLowerCase()` fared a little better with just 193 mismatches, but still.

> Do we need a new op to get a single env var? 

So it sounds like that's the way to go forward indeed.

Another way could be is to somehow extract the collation table that Windows uses. 
I don't know from the top of my head, but I'm sure there's an API for it.